### ISSUE TITLE
Support Transforms

### ISSUE THREAD

**Issue by jakul** [cloudreach/sceptre#421]
When I try to update a stack with Transforms in I get the following error:

"An error occurred (ValidationError) when calling the UpdateStack operation: UpdateStack cannot be used with templates containing Transforms."

This behaviour worked in https://github.com/cloudreach/sceptre/pull/326, which was merged to master in https://github.com/cloudreach/sceptre/commit/10c71e7ab459287a451ccdcfc4876ec97bc5b37d#diff-e2f06afa2a6299964194e9cce31494f4.  That merge was then reverted in https://github.com/cloudreach/sceptre/commit/10c71e7ab459287a451ccdcfc4876ec97bc5b37d#diff-e2f06afa2a6299964194e9cce31494f4 becuase the integration tests were failing. 

@JuanCanham @ngfgrant FYI

**Comment by ngfgrant** [comment]
Thanks for the heads up on this @jakul - will aim to get this back in again soon. Maybe once it is ready to get merged again I could get you to test this from it's branch?

**Comment by jakul** [comment]
Yup sure.

**Comment by JuanCanham** [comment]
yeah, sorry i failed at git, then had to do some other work, i'll try and figure out how to resubmit that at some point this week.

**Comment by ngfgrant** [comment]
Looks like everything is passing as required for this https://github.com/cloudreach/sceptre/pull/419

@jakul maybe you could checkout Juan's changes from the above PR to test:

`git checkout -b JuanCanham-auto-create-change-sets-full-change-for-v1 v1`
`git pull https://github.com/JuanCanham/sceptre.git auto-create-change-sets-full-change-for-v1`

If it is all working I'll get that PR merged in.

**Comment by jakul** [comment]
@JuanCanham https://github.com/cloudreach/sceptre/pull/419 is very far behind `sceptre/master` and I'm unable to test it. Can you update it to have the latest code?

**Comment by jakul** [comment]
Oh. I don't know how to test stuff against v1. I can test against v2 once it's merged in there.

**Comment by SaMnCo** [comment]
Hi there, 

Kudos team for the work on Sceptre. Just started using it and it drastically improved my CF experience. 

I am facing a similar issue as @jakul while working on SAM templates. It seems it has been referenced in here #197, #198, #240 but I don't understand how to get these updates. 
I am currently on 1.4.0 which seems to be the latest and greatest, and I am getting: 

```$ sceptre launch-stack skunkwork cloudsearch-cfcr
[2018-08-16 14:13:17] - sceptre.stack - skunkwork/cloudsearch-cfcr - Launching stack
[2018-08-16 14:13:18] - sceptre.stack - skunkwork/cloudsearch-cfcr - Stack is in the PENDING state
[2018-08-16 14:13:18] - sceptre.stack - skunkwork/cloudsearch-cfcr - Creating stack
An error occurred (ValidationError) when calling the CreateStack operation: CreateStack cannot be used with templates containing Transforms.
```

Any advice / roadmap pointers for this? Also, will sceptre support serverless (the framework) at some point? 

Thanks

**Comment by JuanCanham** [comment]
I'm not sure exactly when it will hit mainline, but given there are MRs, it should be soon (unfortunately due to other commitments @ngfgrant has been left with the unenviable job of figuring out what i did to my git history and how to get the fixes in (sorry Niall))

> Also, will sceptre support serverless (the framework) at some point?

Serverless support is not likely in the short term, as serverless likes to control the cloudformation stacks itself, so we would likely have to have it as a cloud provider in itself and the cfn would become serverless's ~problem~ responsibility (that's not a no though, as I've had a few projects where Serverless+Sceptre would have been a cool, but i think it's lower priority than azure/gcp/etc)

SAM support is something I've already written for 2.0 (as a plugin), so should land in the next few months.

**Comment by SaMnCo** [comment]
Thanks for the quick answer and info, I appreciate it. Still not familiar enough with the tool to assist much at this stage unfortunately. 

I think I'll keep Sceptre + SLS for now. SLS is no good to manage the persistent assets as far as my experience goes, so I still rather deploy the blueprint with Sceptre and only APIG/Lambda with SLS. 

Anyway, thanks for the update, I'll keep an eye on the releases for SAM templates. Again, we appreciate your work, it's awesome.

**Comment by eikenb** [comment]
It looks like the current PR #454 should resolve this and it has been waiting for review since August. Any chance that will get reviewed soon?

**Comment by ngfgrant** [comment]
Resolved in #454
